nameko/containers.py

```diff
@@ -140,22 +156,23 @@ def __init__(self, service_cls, worker_ctx_cls, config):
         self._died = Event()
 
     @property
-    def entrypoints(self):
-        return filter(is_entrypoint_provider, self.dependencies)
+    def extensions(self):
+        return self.entrypoints | self.dependencies | self.subextensions
 
     @property
-    def injections(self):
-        return filter(is_injection_provider, self.dependencies)
+    def interface(self):
+        # TODO: design interface to expose to extensions
+        return self
```

Comment:
this might be a good place to make sure extensions only have a weakref to the container. still investigating whether this is strictly needed, but it might be